### PR TITLE
Commerce 3885 Implement order date filter on orders dataset display

### PR DESCRIPTION
**Pull request by luca-pellizzon**

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/data/set/filter/BaseDateRangeClayDataSetFilter.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ */
+public abstract class BaseDateRangeClayDataSetFilter
+	implements ClayDataSetFilter {
+
+	public abstract DateClayDataSetFilterItem getMax();
+
+	public abstract DateClayDataSetFilterItem getMin();
+
+	public String getPlaceholder() {
+		return "dd/mm/yyyy";
```
Comment (marco-leo) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/data/set/filter/BaseDateRangeClayDataSetFilter.java:28]: @luca-pellizzon this is wrong as it depends on the locale of the user

**Review by marco-leo — CHANGES_REQUESTED** [review]

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/data/set/filter/BaseDateRangeClayDataSetFilter.java (hunk 1 above)
Comment (luca-pellizzon) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/data/set/filter/BaseDateRangeClayDataSetFilter.java:28]: it is not even used, I'm going to remove it
the placeholder is managed by the frontend component

**Review by dsitu — COMMENTED** [review]
SF looks good to me, but @luca-pellizzon @marco-leo do we need to have someone from frontend infra review this also since we are making additions to the frontend-taglib module?

File: modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+			}
+		}
+
+		return new DateClayDataSetFilterItem(day, month, year);
+	}
+
+	@Override
+	public String getType() {
+		return "dateRange";
+	}
```
Comment (riccardo-alberti) [inline modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java:118]: @luca-pellizzon maybe it's better to implement the getType() in the abstract class... so if we need to have another filter for data ranges not for orders we don't need to reimplement this method

Comment (luca-pellizzon) [reply modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java:118]: thanks, I just forgot to move it inside BaseDateRangeClayDataSetFilter, will do it

**Comment by luca-pellizzon** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
sent to https://github.com/liferay-frontend/liferay-portal/pull/549

**Pull request closed without merging**